### PR TITLE
chore: esp-hal-common - update embedded-hal rc

### PR DESCRIPTION
**Pull request by elpiel**
### Must

- [ ] needs a new version of `embassy-time` to be released

- [ ] The code compiles without `errors` or `warnings`.
- [ ] All examples work.
- [ ] `cargo fmt` was run.
- [ ] Your changes were added to the `CHANGELOG.md` in the proper section.
- [ ] You updated existing examples or added examples (if applicable).
- [ ] Added examples are checked in CI

### Nice to have

- [ ] You add a description of your work to this PR.
- [ ] You added proper docs for your newly added features and code.

### CODE REVIEW

File: esp-hal-common/src/gpio.rs [hunk 1]
```diff
@@ -1376,12 +1376,12 @@ impl<MODE> embedded_hal_1::digital::ErrorType for AnyPin<Input<MODE>> {
 
 #[cfg(feature = "eh1")]
 impl<MODE> embedded_hal_1::digital::InputPin for AnyPin<Input<MODE>> {
-    fn is_high(&self) -> Result<bool, Self::Error> {
+    fn is_high(&mut self) -> Result<bool, Self::Error> {
         let inner = &self.inner;
```
Comment (reitermarkus) [inline esp-hal-common/src/gpio.rs:1380]:
```suggestion
        let inner = &mut self.inner;
```

File: esp-hal-common/src/gpio.rs [hunk 2]
```diff
@@ -1376,12 +1376,12 @@ impl<MODE> embedded_hal_1::digital::ErrorType for AnyPin<Input<MODE>> {
...
         handle_gpio_input!(inner, target, { target.is_high() })
     }
 
-    fn is_low(&self) -> Result<bool, Self::Error> {
+    fn is_low(&mut self) -> Result<bool, Self::Error> {
         let inner = &self.inner;
```
Comment (reitermarkus) [inline esp-hal-common/src/gpio.rs:1385]:
```suggestion
        let inner = &mut self.inner;
```

File: esp-hal-common/src/gpio.rs [hunk 3]
```diff
@@ -1442,12 +1442,12 @@ impl<MODE> embedded_hal_1::digital::OutputPin for AnyPin<Output<MODE>> {
 
 #[cfg(feature = "eh1")]
 impl<MODE> embedded_hal_1::digital::StatefulOutputPin for AnyPin<Output<MODE>> {
-    fn is_set_high(&self) -> Result<bool, Self::Error> {
+    fn is_set_high(&mut self) -> Result<bool, Self::Error> {
         let inner = &self.inner;
```
Comment (reitermarkus) [inline esp-hal-common/src/gpio.rs:1446]:
```suggestion
        let inner = &mut self.inner;
```

File: esp-hal-common/src/gpio.rs [hunk 4]
```diff
@@ -1442,12 +1442,12 @@ impl<MODE> embedded_hal_1::digital::OutputPin for AnyPin<Output<MODE>> {
...
         handle_gpio_output!(inner, target, { target.is_set_high() })
     }
 
-    fn is_set_low(&self) -> Result<bool, Self::Error> {
+    fn is_set_low(&mut self) -> Result<bool, Self::Error> {
         let inner = &self.inner;
```
Comment (reitermarkus) [inline esp-hal-common/src/gpio.rs:1451]:
```suggestion
        let inner = &mut self.inner;
```

**Review by reitermarkus — CHANGES_REQUESTED** [review]

**Comment by jessebraham** [comment]
I've taken care of this in #1068, thanks anyway though!

**Pull request closed without merging**